dhis-2/dhis-api/src/main/java/org/hisp/dhis/common/BaseAnalyticalObject.java

```diff
@@ -1214,4 +1217,50 @@ public void setParentGraphMap( Map<String, String> parentGraphMap )
     {
         this.parentGraphMap = parentGraphMap;
     }
+
+    @Override
+    @JsonProperty
+    @JacksonXmlElementWrapper( localName = "subscribers", namespace = DxfNamespaces.DXF_2_0 )
+    @JacksonXmlProperty( localName = "subscriber", namespace = DxfNamespaces.DXF_2_0 )
+    public Set<String> getSubscribers()
+    {
+        return subscribers;
+    }
+
+    public void setSubscribers( Set<String> subscribers )
+    {
+        this.subscribers = subscribers;
+    }
+
+    @Override
+    @JsonProperty
+    @JacksonXmlProperty( namespace = DxfNamespaces.DXF_2_0 )
+    public boolean isSubscribed()
```

Comment:
This doesn't really feel right.. is this for use internally? or to ease the job for the frontend developers? if so, why would they not just do `f.subscribers.includes(userUid)` ? the way its implemented here, we would have a variable on all favourites that would change depending on who is requesting the object.. or do you foresee 1000++ user subscribing to a object?

Reply:
Yes it is to make it easier for consumers / apps I think. We have several fields which change per user e.g. access so I think this is fine.